web/src/components/profile/approved-researcher/AgreementForm.tsx

```diff
@@ -16,15 +15,28 @@ type ApprovedResearcherFormProps = {
 
 export default function ApprovedResearcherForm(props: ApprovedResearcherFormProps) {
   const { agreementId, setAgreementCompleted } = props;
-  const [submitted, setSubmitted] = useState(false);
   const [agreed, setAgreed] = useState(false);
+  const [secondsRemaining, setSecondsRemaining] = useState(approvedAgreementRequiredReadingSeconds);
+  const canAgree = secondsRemaining === 0;
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      setSecondsRemaining((secondsRemaining) => {
+        if (secondsRemaining === 0) {
+          clearInterval(timer);
+          return 0;
+        }
+        return secondsRemaining - 1;
+      });
+    }, 1000);
+  }, []);
```

Comment:
This should have a cleanup function added to avoid memory leaks, something like:

```
  useEffect(() => {
    if (secondsRemaining === 0) return;

    const timer = setTimeout(() => {
      setSecondsRemaining((prev) => prev - 1);
    }, 1000);

    return () => clearTimeout(timer);
  }, [secondsRemaining]);
```